agent/agent-tooling/src/main/java/com/microsoft/applicationinsights/agent/internal/processors/SpanProcessor.java

```diff
@@ -0,0 +1,150 @@
+package com.microsoft.applicationinsights.agent.internal.processors;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+import com.microsoft.applicationinsights.agent.bootstrap.configuration.Configuration.ProcessorConfig;
+import io.opentelemetry.common.AttributeKey;
+import io.opentelemetry.common.Attributes;
+import io.opentelemetry.common.ReadableAttributes;
+import io.opentelemetry.sdk.trace.data.SpanData;
+import org.checkerframework.checker.nullness.qual.Nullable;
+
+
+public class SpanProcessor extends AgentProcessor {
+    private static final Pattern capturingGroupNames = Pattern.compile("\\(\\?<([a-zA-Z][a-zA-Z0-9]*)>");
+    private final List<String> fromAttributes;
+    private final List<String> toAttributeRules;
```

Comment:
do we need to store `toAttributeRules`? looks like we only need the patterns list?